lib/puppet/feature/dsc.rb

```diff
@@ -0,0 +1,20 @@
+require 'puppet/util/feature'
+
+if Puppet.features.microsoft_windows?
+  required_version = Gem::Version.new('5.0.10586.117')
+  installed_version = Gem::Version.new(Facter.value(:powershell_version))
+
+  if (installed_version >= required_version)
+    Puppet.features.add(:dsc)
+  else
+    Puppet.warn_once(
+      'dsc_unavailable',
+      :dsc_unavailable,
+      _("The dsc module requires PowerShell version %{required} - current version %{current}") %
+        { :required => required_version, :current => installed_version},
+      nil,
+      nil,
+      :err
```

Comment:
This is wrong and must be fixed to properly handle Puppet 4 vs Puppet 5 as has been done in https://github.com/puppetlabs/puppetlabs-dsc_lite/pull/72